pkg/common/catalog/configure.go

```diff
@@ -0,0 +1,100 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	spi "github.com/spiffe/spire/proto/spire/common/plugin"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type CoreConfig struct {
+	TrustDomain spiffeid.TrustDomain
+}
+
+func (c CoreConfig) legacy() *spi.ConfigureRequest_GlobalConfig {
+	return &spi.ConfigureRequest_GlobalConfig{
+		TrustDomain: c.TrustDomain.String(),
+	}
+}
+
+func (c CoreConfig) v1() *configv1.CoreConfiguration {
+	return &configv1.CoreConfiguration{
+		TrustDomain: c.TrustDomain.String(),
+	}
+}
+
+type Configurer interface {
+	Configure(ctx context.Context, coreConfig CoreConfig, configuration string) error
+}
+
+type configurerRepo struct {
+	configurer Configurer
+}
+
+func (repo *configurerRepo) Binder() interface{} {
+	return func(configurer Configurer) {
+		repo.configurer = configurer
+	}
+}
+
+func (repo *configurerRepo) Versions() []Version {
+	return []Version{
+		configurerV1Version{},
+	}
+}
+
+func (repo *configurerRepo) Clear() {
```

Comment:
It doesn't seem to be covered in tests.

Reply:
I'm not sure it's worth it. This repo is an internal detail that is never exported and never reused. Clear is never called. I'll leave a comment to that effect.

Reply:
Fair enough.